resources/user-agents/browsers/uc-browser/uc-web-2.json

```diff
@@ -56,58 +62,119 @@
           ]
         },
         {
-          "match": "Mozilla/5.0 (#PLATFORM#) applewebkit* (*khtml*like*gecko*) Version/*Safari/* UCBrowser/#MAJORVER#.#MINORVER#* Mobile*",
+          "match": "Mozilla/5.0 (#PLATFORM##DEVICE# Build/*) applewebkit* (*khtml*like*gecko*) Version/* Safari/* UCBrowser/#MAJORVER#.#MINORVER#*",
           "engine": "WebKit",
+          "devices": {
+            "Galaxy Nexus": "Samsung Galaxy Nexus"
+          },
           "platforms": [
-            "Android_4_3", "Android_4_2", "Android_4_1", "Android_4_0",
-            "Android_2_3",
-            "Android"
+            "Android_4_3", "Android_4_2", "Android_4_1", "Android_4_0"
           ]
         },
         {
-          "match": "UCWEB/*(#PLATFORM#) U2/* UCBrowser/#MAJORVER#.#MINORVER#* U2/*",
-          "device": "general Mobile Phone (Touch)",
-          "engine": "U2",
+          "match": "Mozilla/5.0 (#PLATFORM##DEVICE# Build/*) applewebkit* (*khtml*like*gecko*) Version/* Safari/* UCBrowser/#MAJORVER#.#MINORVER#*",
+          "engine": "WebKit",
+          "devices": {
+            "GT-I9082L": "Samsung GT-I9082L",
+            "HUAWEI P6-U06": "Huawei P6-U06",
+            "Lenovo A516": "Lenovo A516",
+            "LenovoA3300-GV": "Lenovo A3300-GV"
+          },
           "platforms": [
-            "Android_E_4_4", "Android_E_4_3", "Android_E_4_2", "Android_E_4_1", "Android_E_4_0",
-            "Android_E_2_3", "Android_E_2_2", "Android_E_2_1",
-            "Android_E"
+            "Android_4_2"
           ]
         },
         {
-          "match": "UCWEB/*(Linux; U; Opera Mini/*) U2/* UCBrowser/#MAJORVER#.#MINORVER#*",
```

Comment:
This is where most of those Opera Mini patterns are defined that I mention below.